app/assets/javascripts/discourse/tests/acceptance/dismiss-notification-modal-test.js

```diff
@@ -0,0 +1,79 @@
+import { click, visit } from "@ember/test-helpers";
+import {
+  acceptance,
+  exists,
+  query,
+  updateCurrentUser,
+} from "discourse/tests/helpers/qunit-helpers";
+import { test } from "qunit";
+import pretender from "../helpers/create-pretender";
+
+acceptance("Dismiss notification confirmation", function (needs) {
+  needs.user();
+
+  test("does not show modal when no high priority notifications", async function (assert) {
+    pretender.put("/notifications/mark-read", () => {
+      return [200, { "Content-Type": "application/json" }, { success: true }];
+    });
+
+    await visit("/");
+    await click(".current-user");
+    await click(".notifications-dismiss");
+    assert.notOk(exists(".bootbox.modal"));
+  });
+
+  test("shows confirmation modal", async function (assert) {
+    updateCurrentUser({
+      unread_high_priority_notifications: 2,
+    });
+    await visit("/");
+    await click(".current-user");
+    await click(".notifications-dismiss");
+    assert.ok(exists(".bootbox.modal"));
+
+    assert.strictEqual(
+      query(".bootbox.modal .modal-body").innerText,
+      "You have 2 important notifications, are you sure you would like to dismiss?"
+    );
+    await click(".bootbox.modal .btn-default");
```

Comment:
Just a weird test finding when using the bootbox vs a modal:

When creating 2 tests in sequence, using a regular modal doesn't result in test contamination if opening the modals in both tests

But for the same 2 tests, using bootbox results in test contamination for the second test if the first test's bootbox wasn't closed. So we have to be mindful and close it after we're done with it here.